src/hooks/useSnackbar/useSnackbar.test.tsx

```diff
@@ -0,0 +1,279 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { useRef } from 'react';
+
+import { act as rtlAct } from '@testing-library/react';
+import { act, renderHook } from '@testing-library/react-hooks';
+
+import { useSnackbar } from './useSnackbar';
+import { Button } from '../../components/basic/Button';
+import { TIMERS } from '../../components/constants';
+import {
+	CloseSnackbarFn,
+	CreateSnackbarFnArgs,
+	SnackbarManager,
+	SnackbarManagerProps
+} from '../../components/utilities/SnackbarManager';
+import { screen, setup } from '../../test-utils';
+import { ThemeProvider } from '../../theme/theme-context-provider';
+
+jest.mock<typeof import('react-dom')>('react-dom', () => ({
+	...jest.requireActual<typeof import('react-dom')>('react-dom'),
+	createPortal: (node): React.ReactPortal => node as React.ReactPortal
+}));
+
+const snackbarContextError = 'snackbar manager context not initialized';
+beforeEach(() => {
+	const originalErrorFn = console.error;
+	console.error = jest.fn<ReturnType<typeof console.error>, Parameters<typeof console.error>>(
+		(message, ...args) => {
+			// silence error for snackbar
+			if (message !== snackbarContextError) {
+				originalErrorFn(message, ...args);
+			}
+		}
+	);
+});
+
+const Wrapper = ({
+	children,
+	...props
+}: React.PropsWithChildren<SnackbarManagerProps>): React.JSX.Element => (
+	<ThemeProvider>
+		<SnackbarManager {...props}>{children}</SnackbarManager>
+	</ThemeProvider>
+);
+
+const TestComponent = ({ args }: { args: CreateSnackbarFnArgs[] }): React.JSX.Element => {
+	const createSnackbar = useSnackbar();
+	const keyRef = useRef(0);
+	return (
+		<Button
+			onClick={() => {
+				createSnackbar({
+					key: `key-${keyRef.current}`,
+					...args[keyRef.current]
+				});
+				keyRef.current += 1;
+			}}
+			label={'create'}
+		/>
+	);
```

Comment:
Isn't this component really similar to src/hooks/useSnackbar/stories-helpers.tsx? Can this one be reused?

Reply:
They are similar, but the purpose is quite different. This test component has to be able to create different snackbars with different args. The story helper instead is just there because of the props, and it must match the args of the hook function to make the documentation create the args table correctly. Honestly, in the future I hope the stories' helper could be deleted, and the hook could be documented in a cleaner way.
To answer your question, I'd prefer keeping the 2 distinct.

Reply:
Ok, I see